pmlc/conversion/gpu_to_spirv/gpu_to_spirv.cc

```diff
@@ -78,6 +160,16 @@ void populateStdxToSPIRVPatterns(MLIRContext *context,
   patterns.insert<StdxSubgroupBroadcastOpConversion>(context, typeConverter);
 }
 
+void populateStdxToSPIRVGLSLPatterns(MLIRContext *context,
+                                     SPIRVTypeConverter &typeConverter,
+                                     OwningRewritePatternList &patterns) {
+  patterns.insert<StdxRoundOpConversion,
```

Comment:
Changes look good to me, but I am having slight doubts regarding the implementations of the functions that are still missing in spv dialect in mlir upstream. If we are not using the functions from extended instruction sets (GLSL or OpenCL) we can have a gap in performance on those. I would recommend adding those to the upstream first and map them 1:1, later on rely on the gfx compiler during spirv consumption. Probably not needed to do this straightaway, but at least leave the TODO comments to mark those to replace in the future

Reply:
I agree with you that we can leave these ops TODO and replace in the future. This patch is trying to fix these stdx lowing right now to help test networks that needs these ops.

Reply:
I'm not sure if we are in such a hurry to add Round/SinH, etc that it is worth adding to technical debt.  I suspect if you made a PR for upstream to add support for these ops, it would be accepted fairly quickly (and there are all in the GLSL extension as far as I can tell https://www.khronos.org/registry/spir-v/specs/unified1/GLSL.std.450.html).  Let's keep the easy 1-to-1 conversions, but lets fix upstream support for the rest.

Reply:
Thanks. I keeped the 1-to-1 convertions in this PR and will add those ops to upstream.